packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -658,9 +653,37 @@ export abstract class BaseService extends Resource
         this.executeCommandLogConfiguration();
       }
     }
+    // Enable deployment alarms
+    this.enableDeploymentAlarms(props.deploymentAlarms);
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   *   Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization
+  */
+  public enableDeploymentAlarms(alarmConfig?: DeploymentAlarmConfig) {
+    const deploymentConfiguration = this.resource.deploymentConfiguration as CfnService.DeploymentConfigurationProperty;
+    // Throw an error if deployment alarms are already configured
+    if (deploymentConfiguration?.alarms) {
+      throw new Error('Deployment alarms are already configured.');
+    }
+    if (alarmConfig) {
```

Comment:
let's move this logic outside of the function call, similar to how 'enableExecuteCommand' does it. That way it's clear in the constructor code that if there aren't alarms specified in the props, `enableDeploymentAlarms` will not be called.